test/e2e/prometheus_test.go

```diff
@@ -219,9 +219,10 @@ func TestPrometheusRemoteWrite(t *testing.T) {
 `,
 			expected: []remoteWriteTest{
 				{
-					query:       `absent(prometheus_remote_storage_samples_pending{__tmp_openshift_cluster_id__=~".+"})`,
+					query: `absent(prometheus_build_info{__tmp_openshift_cluster_id__=~".+"})`,
+					// absent returns 1 if query is empty
 					expected:    func(v float64) bool { return v == 1 },
-					description: "Expected to find 0 time series of metric prometheus_remote_storage_samples_pending with the temporary cluster_id label",
+					description: "Expected to find 0 time series of metric prometheus_build_info with the temporary cluster_id label, found %[1]s",
```

Comment:
if the test fails, there will be no result hence no `%[1]s`. In fact the description will never be invoked.

Reply:
Right this will only be invoked if the `validate` function fails. In that case there should be a query result already and the descriptions gets wrapped in [WaitForqueryReturn](https://github.com/openshift/cluster-monitoring-operator/blob/d0dad7cb7ce69c7ddd5c1b8dbe9263dd3a644302/test/e2e/framework/client.go#L375). Or am I missing something :thinking:

Reply:
If absent(...) returns something, it will always be 1. Otherwise WaitForQueryReturn will return an error here: https://github.com/openshift/cluster-monitoring-operator/blob/d0dad7cb7ce69c7ddd5c1b8dbe9263dd3a644302/test/e2e/framework/client.go#L371